libraries/AP_Compass/CompassCalibrator.cpp

```diff
@@ -768,6 +768,23 @@ Vector3f CompassCalibrator::calculate_earth_field(CompassSample &sample, enum Ro
     return efield;
 }
 
+/*
+  return true if two rotations are equivalent
+  This copes with the fact that we have some duplicates, like ROLL_180_YAW_90 and PITCH_180_YAW_270
+ */
+bool CompassCalibrator::rotation_equal(enum Rotation r1, enum Rotation r2) const
+{
+    if (r1 == r2) {
+        return true;
+    }
+    Vector3f v(1,2,3);
+    Vector3f v1 = v;
+    Vector3f v2 = v;
+    v1.rotate(r1);
+    v2.rotate(r2);
+    return (v1 - v2).length() < 0.001;
```

Comment:
Why test for the length instead of testing for them being equal?

Reply:
maths rounding due to different order of operations. We need to check that the difference is small, and length is the easiest method.

Reply:
Yeah, that's why I think that a test would be a good idea. They should either run the same math or arrive at a result good enough for `is_equal` to return true (otherwise there is no point for is_equal).

Reply:
is_equal is for very small epsilon. I deliberately want a larger epsilon here, due to the natural differences in rounding for rotations
If you would like to spend the time to prove that is_equal is correct for all rotation cases then go ahead, but I think its just a waste of effort